presto-main/src/main/java/com/facebook/presto/operator/OperatorContext.java

```diff
@@ -239,6 +241,26 @@ public void recordFinish()
         finishUserNanos.getAndAdd(nanosBetween(intervalUserStart.get(), currentThreadUserTime()));
     }
 
+    /**
+     * Get a future that is completed when the {@link #notifyAsync()} method is called, or
+     * next time this method is called.  This api is designed to only be used by {@link Driver}.
+     */
+    ListenableFuture<?> getAsyncNotifyFuture()
```

Comment:
What is the (functional) difference between `setMemoryRevocationRequestListener()` and`getAsyncNotifyFuture()`? (`setMemoryRevocationRequestListener()`  used to be `getMemoryRevokingRequestedFuture`)
I am under impression that this is functionality equivalent. Do we need both?